### ISSUE TITLE
Potential __setup_ui refactor to improve modularity/subclassibility

### ISSUE THREAD

**Issue by KenyonPrater** [niklashenning/pyqttoast#21]
This is a bit subjective, but [this issue I created](https://github.com/niklashenning/pyqttoast/issues/20) is only a github issue (rather than just something I'd implement my own logic for) because `__setup_ui` is a big function without any good ways for a subclass to hook into it. [This issue](https://github.com/niklashenning/pyqttoast/issues/11) is the sort that could also potentially be handled by exposing functions for composition/inheritance.

While it seems like an undertaking, it could potentially be sliced up into smaller functions, so that it's easier for users to make tweaks to the toast widgets.

That said, I do understand the value of a nice clean API that does exactly what you want it to do and doesn't expect the user to go mucking around in its guts causing a mess, so I understand if this is bad design in your view!

**Comment by niklashenning** [comment]
Yeah, I agree with you. Refactoring this method has been on my list for a while now as it's way too big and not clean / intuitive at all. I've been putting this off, but it would be very beneficial for the library, so I think it needs to be done.
